app/services/concerns/xccdf/rule_results.rb

```diff
@@ -33,11 +33,15 @@ def selected_op_rule_results
       end
     end
 
+    def test_result_rules_missing
+      selected_op_rule_results.map(&:id) - rule_ids.keys
```

Comment:
This rather seems to check extra rules present in the test result rather than missing rules in a test result.

Reply:
Well, think about the failure conditions: if there are extra rules present in the test result, we can't parse it because our benchmark doesn't know those rules. If there are missing rules in the test result, that's OK because some rules may not be applicable or selected. We can still ingest the report in that case, and we shouldn't expect errors in later parsing steps (i.e. RuleResult saving against a non-existent Rule record).

Reply:
Can it filter the subset of rules from those that it already knows, and report it as unsupported? It would create a new scenario.

Would you mind renaming it to something like `test_result_extra_rules`?

Reply:
The name of the method matches the error message displayed:

```
UnknownRuleError: The following rules are missing from profile xccdf_org.ssgproject.content_profile_standard in benchmark xccdf_org.ssgproject.content_benchmark_RHEL-8 with SSG
 0.1.40:                                                                                                 
xccdf_org.ssgproject.content_rule_sshd_set_keepalive
...
xccdf_org.ssgproject.content_rule_grub2_nousb_argument.rule
Report for profile xccdf_org.ssgproject.content_profile_standard against lenovolobato.lobatolan.home of account 0 could not be parsed.
```

I will rename it to `test_result_rules_unknown` to match the error class rather than the message.